src/lib.rs

```diff
@@ -87,108 +106,169 @@ impl DstService {
+                        .map(|(addr, EndpointMeta { h2 })| {
+                            let protocol_hint = if *h2 {
+                                Some(pb::ProtocolHint {
+                                    protocol: Some(pb::protocol_hint::Protocol::H2(
+                                        pb::protocol_hint::H2::default(),
+                                    )),
+                                })
+                            } else {
+                                None
+                            };
+                            pb::WeightedAddr {
+                                addr: Some(addr.into()),
+                                protocol_hint,
+                                ..Default::default()
+                            }
+                        })
+                        .collect::<Vec<_>>();
+                    if !added.is_empty() {
+                        tracing::debug!(?added);
+                        tx.send(Ok(pb::Update {
+                            update: Some(pb::update::Update::Add(pb::WeightedAddrSet {
+                                addrs: added,
+                                ..Default::default()
+                            })),
+                        }))
+                        .await?;
+                    }
+
+                    let removed = prev
+                        .keys()
+                        .filter(|addr| !curr.contains_key(addr))
+                        .map(Into::into)
+                        .collect::<Vec<_>>();
+                    if !removed.is_empty() {
+                        tracing::debug!(?removed);
+                        tx.send(Ok(pb::Update {
+                            update: Some(pb::update::Update::Remove(pb::AddrSet {
+                                addrs: removed,
+                            })),
+                        }))
+                        .await?;
                     }
-                    tracing::debug!("Watch ended");
-                    Ok(())
+
+                    prev = curr;
                 }
-                .map_err(|_: mpsc::error::SendError<_>| tracing::info!("Lookup closed"))
-                .in_current_span(),
-            );
-            rx
-        } else {
-            tracing::info!("Does not exist");
-            let (mut tx, rx) = mpsc::channel(1);
-            let _ = tx
-                .send(Ok(pb::Update {
-                    update: Some(pb::update::Update::NoEndpoints(pb::NoEndpoints {
-                        exists: false,
-                    })),
-                }))
-                .await;
-            rx
-        }
+                tracing::debug!("Watch ended");
+                Ok(())
+            }
+            .map_err(|_: mpsc::error::SendError<_>| tracing::info!("Lookup closed"))
+            .in_current_span(),
+        );
+
+        rx
+    }
+
+    #[tracing::instrument(skip(self), level = "info")]
+    async fn stream_overrides(
```

Comment:
It could be. For now, I find it clearer that everything is clearly labeled as either endpoints or overrides. This would all need quite a bit of changing to support arbitrary profiles